corehq/apps/accounting/tests/utils.py

```diff
@@ -54,3 +62,31 @@ def teardown_subscription(cls, domain):
             cls.__accounts[domain].delete()
         finally:
             Subscription._get_active_subscription_by_domain.clear(Subscription, domain)
+
+
+@contextmanager
+def mocked_stripe_api_context(*args, **kwargs):
+    with mock.patch.object(stripe.Customer, 'modify_source') as mock_modify_source, \
+         mock.patch.object(stripe.Customer, 'list_sources') as mock_list_sources:
+        setup_stripe_common_mocks(mock_list_sources, mock_modify_source)
+        yield
+
+
+def mocked_stripe_api_decorator(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        with mocked_stripe_api_context():
+            return func(*args, **kwargs)
+    return wrapper
```

Comment:
I think you could define a `ContextDecorator` class named `mock_stripe_api` which can act as a both a context manager and decorator. (docs [here](https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator))

Reply:
A `contextmanager` is a `ContextDecorator`:

> [contextmanager()](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager) uses [ContextDecorator](https://docs.python.org/3/library/contextlib.html#contextlib.ContextDecorator) so the context managers it creates can be used as decorators as well as in [with](https://docs.python.org/3/reference/compound_stmts.html#with) statements.

Reply:
What accept varargs in the context manager? They appear to be unused.
```suggestion
@contextmanager
def mocked_stripe_api():
    with mock.patch.object(stripe.Customer, 'modify_source') as mock_modify_source, \
         mock.patch.object(stripe.Customer, 'list_sources') as mock_list_sources:
        setup_stripe_common_mocks(mock_list_sources, mock_modify_source)
        yield
```

Reply:
Hi Daniel, I initially started with the code as you suggested, then I got `TypeError: mocked_stripe_api() takes 0 positional arguments but 1 was given`... so I let it accept args...

Reply:
I also get `TypeError: Can't make a test from <contextlib._GeneratorContextManager object at 0x111b598b0>` if I don't wrap the Context Manager with a Decorator

Reply:
See https://github.com/dimagi/commcare-hq/pull/33775#discussion_r1419544709